commit b137449b61320e6c466c68aaa2b5bcf7617eb30b (refs/changes/04/1204/1)
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2017-01-04T10:18:17-08:00 (2 years, 9 months ago)
    
    NET-368 NAPI resolvers should not be shuffled for booting CNs

diff --git a/lib/bootparams.js b/lib/bootparams.js
index 33c619f..47e01ba 100644
--- a/lib/bootparams.js
+++ b/lib/bootparams.js
@@ -97,24 +97,6 @@ function getFromCache(opts, callback) {
 }
 
 
-/**
- * Shuffle the contents of the array based on the Fisher-Yates or Knuth shuffle.
- */
-function shuffle(array) {
-    var m = array.length, t, i;
-
-    while (m) {
-        i = Math.floor(Math.random() * m--);
-        t = array[m];
-        array[m] = array[i];
-        array[i] = t;
-    }
-
-    return array;
-}
-
-
-
 // --- Exported functions
 
 
@@ -321,7 +303,7 @@ function getBootParams(opts, callback) {
 
         params.ip = bootNic.ip;
         params.netmask = bootNic.netmask;
-        params.resolvers = shuffle(bootNic.resolvers);
+        params.resolvers = bootNic.resolvers;
 
         // Allow kernel_args from CNAPI to override the nic tag values, but
         // dutifully complain about it
